Builder/Program.cs

```diff
@@ -0,0 +1,163 @@
+        }
+        public WhitePizze GetPizza()
+        {
+            WhitePizze result = this.WhitePizze;
+            this.Reset();
+            return result;
+        }
+    }
+    public class WheatPizzaBuilder : IPizzaBuilder
+    {
+        private WheatPizza WheatPizza = new WheatPizza();
+
+        public WheatPizzaBuilder()
+        {
+            this.Reset();
+        }
+        public void Reset()
+        {
+            this.WheatPizza = new WheatPizza();
+        }
+        public void BuildDough()
+        {
+            this.WheatPizza.AddIngredient("Crunchy Dough");
+        }
+        public void BuildSauce()
+        {
+            this.WheatPizza.AddIngredient("Extra Sauce");
+        }
+        public void BuildTopping()
+        {
+            this.WheatPizza.AddIngredient("Little Toppings");
+        }
+        public WheatPizza GetPizza()
+        {
+            WheatPizza final = this.WheatPizza;
+            this.Reset();
+            return final;
+        }
+    }
+    public class WhitePizze
+    {
+        private List<object> Ingredients = new List<object>();
+        public void AddIngredient(object Ingredient)
+        {
+            Ingredients.Add(Ingredient);
+        }
+        public string ListIngredients()
+        {
+            string str = "";
+            for (int i = 0; i < this.Ingredients.Count; i++)
+            {
+                str += this.Ingredients[i] + "\n";
+            }
+            return "White Pizza order : \n" + str;
+        }
+    }
+    public class WheatPizza
+    {
+        private List<object> Ingredients = new List<object>();
+        public void AddIngredient(object Ingredient)
+        {
+            Ingredients.Add(Ingredient);
+        }
+        public string ListIngredients()
+        {
+            string str = "";
+            for (int i = 0; i < this.Ingredients.Count; i++)
+            {
+                str += this.Ingredients[i] + "\n";
+            }
+            return "Wheat Pizza order : \n" + str;
+        }
+
+    }
+
+    public class Director
+    {
+        private IPizzaBuilder _pizzaBuilder;
+        public IPizzaBuilder PizzaBuilder
+        {
+            set { this._pizzaBuilder = value; }
+        }
+        public void makePizza()
+        {
+            this._pizzaBuilder.BuildDough();
+            this._pizzaBuilder.BuildSauce();
+            this._pizzaBuilder.BuildTopping();
+        }
+        public void makePizzaWithoutToppings()
+        {
+            this._pizzaBuilder.BuildDough();
+            this._pizzaBuilder.BuildSauce();
+        }
+    }
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            // Main objectives
+            Director director = new Director();
+            WheatPizzaBuilder Wheat = new WheatPizzaBuilder();
+            WhitePizzeBuilder white = new WhitePizzeBuilder();
+            WhitePizzeBuilder white2 = new WhitePizzeBuilder();

```

Comment:
You can reuse `white` without declearing `white2` because you have the `reset` method which allows you to use the same builder without worrying about mixing the previous order with the new one .
> try your code with commenting `reset` method and use the same builder to see the difference